tests/end-to-end/director/labeling_api_test.go

```diff
@@ -0,0 +1,845 @@
+						%s
+					}
+				}`,
+			firstApp.ID, labelKey, validLabelValue, tc.gqlFieldsProvider.ForApplicationLabel()))
+	label := graphql.Label{}
+
+	// WHEN
+	err = tc.RunQuery(ctx, setLabelRequest, &label)
+
+	//THEN
+	require.NoError(t, err)
+	require.NotEmpty(t, label.Key)
+	require.NotEmpty(t, label.Value)
+
+	// Try to delete labeldefinition when it's used by some labels
+	// GIVEN
+	queryAppReq := gcli.NewRequest(
+		fmt.Sprintf(`query {
+			result: deleteLabelDefinition(key: "%s") {
+					%s
+				}
+			}`, labelKey, tc.gqlFieldsProvider.ForLabelDefinition()))
+
+	// WHEN
+	err = tc.RunQuery(context.Background(), queryAppReq, nil)
+
+	//THEN
+	require.Error(t, err)
+
+	// Try to delete labeldefinition with deleting related labels
+	// GIVEN
+	queryAppReq = gcli.NewRequest(
+		fmt.Sprintf(`query {
+			result: deleteLabelDefinition(key: "%s", deleteRelatedLabels: %s) {
+					%s
+				}
+			}`, labelKey, "true", tc.gqlFieldsProvider.ForLabelDefinition()))
+
+	// WHEN
+	err = tc.RunQuery(context.Background(), queryAppReq, nil)
+
+	//THEN
+	require.NoError(t, err)
+
+	//Check if label was deleted from application
+	getApp := gcli.NewRequest(
+		fmt.Sprintf(`query {
+				result: application(id: %s,) {
+						%s
+					}
+				}`,
+			firstApp.ID, tc.gqlFieldsProvider.ForApplication()))
+
+	app := graphql.Application{}
+	// WHEN
+	err = tc.RunQuery(ctx, getApp, app)
+
+	//THEN
+	require.NoError(t, err)
+	require.NotEmpty(t, app)
+	assert.NotContains(t, app.Labels, labelKey)
+
+	// Check if LabelDefinition was deleted
+	// GIVEN
+	getLabelDefinitionRequest := gcli.NewRequest(
+		fmt.Sprintf(`query {
+				result: labelDefinition(key: %s) {
+						%s
+					}
+				}`,
+			labelKey, tc.gqlFieldsProvider.ForLabelDefinition()))
+
+	// WHEN
+	err = tc.RunQuery(ctx, getLabelDefinitionRequest, nil)
+
+	//THEN
+	require.Error(t, err)
+}
+
+func TestDeleteScenarioLabel(t *testing.T) {
+	//Create application
+	// GIVEN
+	ctx := context.Background()
+	in := generateSampleApplicationInput("first")
+	appInputGQL, err := tc.graphqlizer.ApplicationInputToGQL(in)
+	require.NoError(t, err)
+	createReq := gcli.NewRequest(
+		fmt.Sprintf(`mutation {
+  				result: createApplication(in: %s) {
+    					%s
+					}
+				}`, appInputGQL, tc.gqlFieldsProvider.ForApplication()))
+	firstApp := graphql.Application{}
+
+	//WHEN
+	err = tc.RunQuery(ctx, createReq, &firstApp)
+
+	//THEN
+	require.NoError(t, err)
+	require.NotEmpty(t, firstApp.ID)
+	defer deleteApplication(t, firstApp.ID)
+
+	//Try to delete scenario label on application
+	// GIVEN
+	labelKey := "scenario"
+	deleteAppLabelRequest := gcli.NewRequest(
+		fmt.Sprintf(`mutation {
+				result: deleteApplicationLabel(applicationID: %s, key: %s) {
+						%s
+					}
+				}`,
+			firstApp.ID, labelKey, tc.gqlFieldsProvider.ForApplicationLabel()))
+
+	// WHEN
+	err = tc.RunQuery(ctx, deleteAppLabelRequest, nil)
+
+	//THEN
+	require.Error(t, err)
+}
+
+func TestDeleteDefaultValueInScenarioLabel(t *testing.T) {
```

Comment:
What about deleting scenarios LabelDefinition? It should be also impossible, right? This is open question.

Reply:
added test for this behaviour